queries/jsonnet/highlights.scm

```diff
@@ -30,63 +65,100 @@
   ","
   ";"
   ":"
-  "::"
-  ":::"
 ] @punctuation.delimiter
 
-; Nested field inheritance
-("+" @punctuation.delimiter [":" "::" ":::"])
-
-"=" @operator
-(unaryop) @operator
-(binary
-  operator: (_) @operator)
-
-"for" @repeat
-
-"function" @keyword.function
+[
+  "::"
+  ":::"
+] @punctuation.special
 
-"in" @keyword.operator
+(field
+  (fieldname) "+" @punctuation.special)
 
+; Imports
 [
- (local)
- (tailstrict)
- "assert"
-] @keyword
+  (import)
+  (importstr)
+] @include
 
-[
-  "else"
-  "if"
-  "then"
-] @conditional
+; References
 
-"error" @exception
+; Make reference same color as parameter 
+; (may incur performance issues on big files)
+; Depends on locals.scm
+((id) @parameter.reference
+ (#is? @parameter.reference parameter))
 
-[
-  (dollar)
-  (self)
-  (super)
-] @variable.builtin
-((id) @variable.builtin
- (#eq? @variable.builtin "std"))
+((id) @function.reference
+ (#is? @function.reference function))
+
+;((id) @var.reference
+; (#is? @var.reference var))
+;((id) @define
+; (#is? @var.reference var))
```

Comment:
I've commented this out now, any clue on how I can get references for things that aren't functions/parameters?

Reply:
You can't get references with treesitter highlighting (in the sense of LSP, if I understand you correctly). You'd need locals queries for that, but Neovim does not support that for highlighting (and upstream has soured on that as well).

Reply:
Mind running the CI checks?

Reply:
Thanks!! ;)

Reply:
I've removed the references part from highlights, it was working but quite hacky, I'll put it in the nvim-jsonnet plugin so we can iterate at a different pace.